src/harness/fourslashImpl.ts

```diff
@@ -3481,8 +3481,19 @@ namespace FourSlash {
 
         public verifyRefactorAvailable(negative: boolean, triggerReason: ts.RefactorTriggerReason, name: string, actionName?: string, actionDescription?: string) {
             let refactors = this.getApplicableRefactorsAtSelection(triggerReason);
-            refactors = refactors.filter(r =>
-                r.name === name && (actionName === undefined || r.actions.some(a => a.name === actionName)) && (actionDescription === undefined || r.actions.some(a => a.description === actionDescription)));
+            refactors = refactors.filter(r => r.name === name);
+
+            refactors = actionName === undefined ? refactors : refactors.map(r => {
+                                r.actions = r.actions.filter(a => a.name === actionName);
+                                return r;
+                            });

```

Comment:
I tried to come up with an all-in-one expression like before and I couldn't, at least without a complicated `map` followed a by a complicated `filter`. I think splitting it out like this is better. But it would be less surprising to use `forEach` instead of `map` since there's a side effect. And you won't have to `return r`.

Actually, writing imperative code might be even better, like

```suggestion
            if (actionName) {
                for (const r of refactors) {
                    r.actions = r.actions.filter(a => a.name === actionName);
                }
            }
```

Since that's the usual way to write side effects in JS.

Reply:
I’m actually not seeing what was wrong with the original one-liner now, though I recall @iisaduan describing the effect and it sounding wrong. Remind me what the problem was?

Reply:
@andrewbranch When calling to check if there's a refactor with `actionName` and `actionDescription`, it feels like it should return `true` if and only if there is a single refactor with both the given name and description. However, it would return true if there were two different refactors, one of which has the name, and the other with the description.

Reply:
Ah, so maybe the two `.some`s could have just been combined. I would personally probably still try to make this a single `filter`, but I’m not too picky about how code is written in the fourslash implementation.